packages/api-docs-builder/buildApiUtils.ts

```diff
@@ -9,9 +9,17 @@ import { replaceComponentLinks } from './utils/replaceUrl';
 import findPagesMarkdownNew from './utils/findPagesMarkdown';
 import { TypeScriptProject } from './utils/createTypeScriptProject';
 
-const systemComponents = fs
-  .readdirSync(path.resolve('packages', 'mui-system', 'src'))
-  .filter((pathname) => pathname.match(/^[A-Z][a-zA-Z]+$/));
+let systemComponents: string[] | undefined;
+// making the resolution lazy to avoid issues when importing something irrelevant from this file (i.e. `getSymbolDescription`)
+// the eager resolution results in errors when consuming externally (i.e. `mui-x`)
+function getSystemComponents() {
```

Comment:
Fixes an issue caused by https://github.com/mui/material-ui/pull/36104/files#diff-62bd94d50324a1a55e28a5677cfd0731f5d40173bca190968d2fba05bbfbec49
Which results in the following error when running on `mui-x` even after adding `@mui-internal/docs-utilities` module resolution:
```sh
yarn docs:api
yarn run v1.22.19
$ yarn docs:api:build
$ cross-env BABEL_ENV=development babel-node -i "/node_modules/(?!@mui)/" -x .ts,.tsx,.js ./docs/scripts/api/buildApi.ts
node:internal/fs/utils:347
    throw err;
    ^

Error: ENOENT: no such file or directory, scandir '/Users/lukas/Documents/dev/mui/mui-x/packages/mui-system/src'
    at Object.readdirSync (node:fs:1438:3)
    at Object.<anonymous> (/Users/lukas/Documents/dev/mui/mui-x/node_modules/@mui/monorepo/packages/api-docs-builder/buildApiUtils.ts:13:4)
```

Reply:
I've [extracted this and other changes](https://github.com/mui/material-ui/pull/36310) unrelated to the original PR idea.